libfprint/drivers/goodixtls/5395/goodix_device.c

```diff
@@ -258,3 +265,114 @@ gboolean fpi_goodix_device_reset(FpDevice *dev, guint8 reset_type, gboolean irq_
     GError *error = NULL;
     return fpi_goodix_device_send(dev, message, TRUE, 500, FALSE, &error);
 }
+
+gboolean fpi_goodix_device_gtls_connection(FpDevice *dev, GError *error)
+{
+    fp_dbg("fpi_goodix_device_gtls_connection()");
+    fp_dbg("Starting GTLS handshake");
+    FpiGoodixDevice *self = FPI_GOODIX_DEVICE(dev);
+    FpiGoodixDevicePrivate *priv = fpi_goodix_device_get_instance_private(self);
+    priv->gtls_params = fpi_goodix_device_gtls_init_params();
+    fpi_goodix_device_gtls_client_hello_step(dev);
+    if (!fpi_goodix_device_gtls_server_identity_step(dev, error))
+    {
+        return FALSE;
+    }
+    // fpi_goodix_gtls_server_done_step();
+    fp_dbg("GTLS handshake successful");
+    return TRUE;
+}
+
+GoodixGTLSParams *fpi_goodix_device_gtls_init_params(void)
+{
+    GoodixGTLSParams *params = g_malloc0(sizeof(GoodixGTLSParams));
+    params->state = 0;
+    params->server_random = g_byte_array_new();
+    params->server_identity = g_byte_array_new();
+    return params;
+}
+
+void fpi_goodix_device_gtls_client_hello_step(FpDevice *dev)
```

Comment:
Needs to check command execution status

Reply:
What about that? Here I meant that we need handle error in this function. I will rewrite it to next:

gboolean fpi_goodix_device_gtls_client_hello_step(FpDevice *dev, GError **error). Return the status (success, failure) from method and if error occurs - create GError **error object with error description